storage/src/context.rs

```diff
@@ -290,22 +270,73 @@ impl ContextApi for TezedgeContext {
         let merkle = self.merkle.read().expect("lock poisoning");
         merkle.get_staged_root_hash()
     }
+
+    fn block_applied(&self) -> Result<(), ContextError> {
+        let mut merkle = self.merkle.write()?;
+        Ok(merkle.block_applied()?)
+    }
+
+    fn cycle_started(&self) -> Result<(), ContextError> {
+        let mut merkle = self.merkle.write()?;
+        Ok(merkle.start_new_cycle()?)
+    }
+
+    fn get_memory_usage(&self) -> Result<usize, ContextError> {
+        let merkle = self.merkle.write()?;
+        Ok(merkle.get_memory_usage()?)
+    }
 }
 
 // context implementation using merkle-tree-like storage
 #[derive(Clone)]
 pub struct TezedgeContext {
-    block_storage: BlockStorage,
+    block_storage: Option<BlockStorage>,
     merkle: Arc<RwLock<MerkleStorage>>,
 }
 
 impl TezedgeContext {
-    pub fn new(block_storage: BlockStorage, merkle: Arc<RwLock<MerkleStorage>>) -> Self {
+    pub fn new(block_storage: Option<BlockStorage>, merkle: Arc<RwLock<MerkleStorage>>) -> Self {
         TezedgeContext {
             block_storage,
             merkle,
         }
     }
+
+    fn associate_block_and_context_hash(
```

Comment:
@mat13mn @tizoc @jurajselep 
ok, so last thing, lets discuss and finally clean this old-hack: https://simplestakingcom.atlassian.net/browse/TE-447